### PR TITLE
Makefile.template: fixed variables names, and added a missing one

### PR DESCRIPTION
**Pull request by emmenlau**
Required for MINGW or MSYS2 compilation

### CODE REVIEW

**Comment by jtattermusch** [comment]
Can you please elaborate the description of the PR for future reference?

**Comment by jtattermusch** [comment]
this is ok to test

File: templates/Makefile.template [hunk 1]
```diff
@@ -1303,7 +1303,7 @@
   % if not lib.get('external_deps', None):
   	$(E) "[INSTALL] Installing $(SHARED_PREFIX)${lib.name}$(SHARED_VERSION_${lang_to_var[lib.language]}).$(SHARED_EXT_${lang_to_var[lib.language]})"
   	$(Q) $(INSTALL) -d $(prefix)/lib
-  	$(Q) $(INSTALL) $(LIBDIR)/$(CONFIG)/$(SHARED_PREFIX)${lib.name}$(SHARED_VERSION_${lang_to_var[lib.language]}).$(SHARED_EXT_${lang_to_var[lib.language]}) $(prefix)/lib/$(SHARED_PREFIX)${lib.name}$(SHARED_VERSION).$(SHARED_EXT_${lang_to_var[lib.language]})
+  	$(Q) $(INSTALL) $(LIBDIR)/$(CONFIG)/$(SHARED_PREFIX)${lib.name}$(SHARED_VERSION_${lang_to_var[lib.language]}).$(SHARED_EXT_${lang_to_var[lib.language]}) $(prefix)/lib/$(SHARED_PREFIX)${lib.name}$(SHARED_VERSION_${lang_to_var[lib.language]}).$(SHARED_EXT_${lang_to_var[lib.language]})
```
Comment (jtattermusch) [inline templates/Makefile.template]: Did you forget to include changes to the regenerated Makefile (only changing template here).

File: templates/Makefile.template [hunk 2]
```diff
@@ -1558,11 +1558,11 @@
   	$(E) "[LD]      Linking $@"
   	$(Q) mkdir -p `dirname $@`
   ifeq ($(SYSTEM),Darwin)
-  	$(Q) ${ld} ${ldflags} -L$(LIBDIR)/$(CONFIG) -install_name $(SHARED_PREFIX)${lib.name}$(SHARED_VERSION).$(SHARED_EXT_${lang_to_var[lib.language]}) -dynamiclib -o ${out_libbase}.$(SHARED_EXT_${lang_to_var[lib.language]}) ${common}${link_libs}
+  	$(Q) ${ld} ${ldflags} -L$(LIBDIR)/$(CONFIG) -install_name $(SHARED_PREFIX)${lib.name}$(SHARED_VERSION_${lang_to_var[lib.language]}).$(SHARED_EXT_${lang_to_var[lib.language]}) -dynamiclib -o ${out_libbase}.$(SHARED_EXT_${lang_to_var[lib.language]}) ${common}${link_libs}
```
Comment (jtattermusch) [inline templates/Makefile.template]: Why change things for Darwin? I am not sure I follow.

**Comment by emmenlau** [comment]
Sorry I should have elaborated more! The Makefile.template references the variable
`$(SHARED_VERSION)` that AFAIK is undefined. The variable AFAIK should be
correctly named `$(SHARED_VERSION_${lang_to_var[lib.language]})`. I've compared
the Makefile to older versions and I think `$(SHARED_VERSION)` has been
deprecated, but was not consistently replaced. This is just a guess though.

However the whole discussion is a bit academic, since on most platforms the variables
`$(SHARED_VERSION_XXX)` are defined to empty. So its just a fix for consistency, it
should not actually change the build result.

Finally in line 1523 the `$(SHARED_VERSION_' + lang_to_var[dep_lib.language] + ')`
was missing for mingw. I added it there.

File: templates/Makefile.template (hunk 1 above)
Comment (emmenlau) [inline templates/Makefile.template]: Yes, I'm sorry that I was not able to correctly create a Makefile from the Makefile.template. Thanks to your
hint (installing golang in Ubuntu) it works now. Should I add the changes to the Makefile?

File: templates/Makefile.template (hunk 2 above)
Comment (jtattermusch) [inline templates/Makefile.template]: Got it based on the PR description.

**Comment by jtattermusch** [comment]
Known test failures:
#9150
#8989

**Pull request merged**